src/app/services/file-builder/file-builder.servise.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { UploadResult } from '../upload/upload-result.interface';
+
+@Injectable()
+export class FileBuilderService {
+
+  getPublicUrl(file: UploadResult, fileUrlPath: string): string {
+    return fileUrlPath + file.cleanName;
+  }
+
+  getFilePath(file: UploadResult, filePrefix: string): string {
+    return filePrefix + file.cleanName;
+  }
+
+  buildResult(file: File): UploadResult {
+    const cleanFilename: string = this.cleanFilename(file.name);
+    return {
+      id: this.generateID(),
```

Comment:
this is not a correct UDID but it will generate a ID like `1072663250-2188118301-4268473884-3687350452`